File: web/src/types/report.ts
export interface ReportInfo {
  title: string;
  project: string;
  company: string;
  author: string;
  version: string;
}

export interface Report extends ReportInfo {
  id: string;
  status: string;
  pdf_path?: string;
  created_at: string;
  updated_at: string;
}

export interface ReportUploadInput {
  id: string;
  title: string;
  description: string;
  notes: string;
  order_index: number;
}

export interface CreateReportRequest extends ReportInfo {
  uploads: ReportUploadInput[];
}

File: internal/models/report.go
package models

import "time"

type Report struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Project     string    `json:"project"`
	Company     string    `json:"company"`
	Author      string    `json:"author"`
	Version     string    `json:"version"`
	Status      string    `json:"status"`
	PDFPath     string    `json:"pdf_path,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ReportInfo struct {
	Title   string `json:"title"`
	Project string `json:"project"`
	Company string `json:"company"`
	Author  string `json:"author"`
	Version string `json:"version"`
}
